app/models/certificate_expiry_reminder.rb

```diff
@@ -0,0 +1,73 @@
+require 'notify/notification'
+require 'auth/authentication_backend'
+
+module CertificateExpiryReminder
+  class Check
+    include Notification
+    include AuthenticationBackend
+
+    REMIND_DAYS_LEFT = [30, 7, 3].freeze
+
+    def run
+      Rails.logger.info("Looking for expiring certs which are enabled...")
+      certificates = expiring_certs
+      Rails.logger.info("Found #{certificates.count} expiring certs.")
+      if certificates.any?
+        send_notifications(group_by_team(certificates))
+      end
+    end
+
+  private
+
+    def expiring_certs
+      Certificate.where('enabled = ?', true).select { |c| REMIND_DAYS_LEFT.include?(c.days_left) && !c.component.nil? }
+    end
+
+    def group_by_team(certificates)
+      certificates.group_by { |c| c.component.team_id }
+    end
+
+    def send_notifications(team_certificates)
+      team_certificates.each do |team_id, certificates|
+        team = Team.find_by_id(team_id)
+        certs_by_days_left = certificates.group_by(&:days_left)
+        certs_by_days_left.each do |days, certs|
+          send_reminder(team, certs, days)
+        end
+      end
+    end
+
+    def send_reminder(team, certs, days)
+      recipients = team_recipients(team.team_alias)
+      certificates = certs.sort_by { |cert| [cert.component.environment, cert.component_type] }.reverse
+      certificates_list = certificates.map do |cert|
+        "#{component_name(cert)} (#{cert.component.environment}): #{cert.usage} certificate - expires on #{cert.x509.not_after}"
+      end
+      Rails.logger.info("Sending a #{days}-day reminder email to #{team.name} team (#{recipients.count} recipients, #{certificates_list.count} certificates)")
+      Rails.logger.error("No recipients found for #{team.name}!") if recipients.empty?
+      recipients.each do |email|
+        send_reminder_email(
+          email_address: email,
+          team_name: team.name,
+          days_left: days,
+          certificates: certificates_list,
+        )
+      end
+    end
+
+    def team_recipients(team_alias)
+      users = get_users_in_group(group_name: team_alias)
+      users.map { |user| user.attributes.find { |att| att.name == 'email' }.value }
+    end
+
+    def component_name(cert)
```

Comment:
This is fine, I just think component has something doing similar logic, so wonder if we could do ```  cert.component.display ``` to reuse the logic

Reply:
It's returning different values to what we want... for the email we want full names, the `display` method return the short names, i.e. VSP ... but yeah, i can add a new method on the component

Reply:
Ah I see, display returns I18n.t("user_journey.component_long_name.#{type}") instead of I18n.t("user_journey.component_long_name.#{type}").  

I guess you could but I don't think there many instances of where it can be reused. So am fine with this

Reply:
I've already made the change in https://github.com/alphagov/verify-self-service/pull/328/commits/9a7c6f38f7f3c9c8791f70af61482b12c4fdefc8 :)

Reply:
That's actually nicer